### DISCUSSION TITLE
commands not found

### DISCUSSION THREAD

**Discussion by mattdimicelli** [ohmyzsh/ohmyzsh#9736] [Help] [answered]
![versions](https://user-images.githubusercontent.com/67718737/111224484-e6ef9300-85b4-11eb-8546-f0fc7afaf46b.png)
Hi.  I'm pretty new to Linux and the command line in general, so please go easy on me!  

Yesterday I upgraded to ZSH from bash on Xubuntu Linux.  At first I tried 
sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)", 
but it told me that I had to install ZSH first, so I installed ZSH via sudo apt install zsh.  Afterwards, I was able to sucessfully install OhmyZSH.  

Pretty much immediately I noticed that I was having issues with certain commands.  For example, I am able to do "node --version" and get the response, but when I try to run npm (which is installed with node) I get :

zsh: command not found: npm

I have since installed a few programs via the command line, and I am getting the same type of errors when trying to run them. 

I know that this probably has to do with my path variable, and I read up about it and have tried to understand it, but I still feel over my head.  I haven't been able to get this working.

I'm also getting different versions of the same programs, depending on which shell I run it from!  I don't understand the mechanics behind this...

Any help would be GREATLY appreciated!  Thank you!!

**Comment by mcornella** [comment, marked as answer]
zsh uses another rc file which you can find at `~/.zshrc` (same as `$HOME/.zshrc`, if you put that in the terminal you'll find it). It seems that you had a `$PATH` configuration, probably in `.bashrc` or `.profile` (these files are also in `$HOME`). So look for lines changing `$PATH` and copy them to the `.zshrc` file, at the top of the file.

**Reply by mattdimicelli** [reply]
Hi Mcornella,
My .bashrc profile does not reference the path variable.
My . profile shows the following:

```
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" ] ; then
    PATH="$HOME/bin:$PATH"
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi

This is what I have in my .zshrc file:
# If you come from bash you might have to change your $PATH.
# export PATH=$HOME/bin:/usr/local/bin:$PATH

PATH=/bin:/usr/bin:/usr/local/bin:${PATH}
export PATH
```

I'm sorry if I'm being obtuse, but can you tell me exactly how my .zshrc file should look here?

**Reply by mcornella** [reply]
Does npm work inside bash?

**Reply by mattdimicelli** [reply]
Yes, it does.  I actually just editted my first post in the chain with a screenshot, take a look at it.

**Reply by mcornella** [reply]
Ok can you run `which npm` in bash and post the results?

**Reply by mattdimicelli** [reply]
Yes.  I ran `which npm ` and got the following: `/home/mrd2689a/.nvm/versions/node/v15.8.0/bin/npm`

**Reply by mcornella** [reply]
Oh you're using nvm, you need to set it up under zsh as well. In OMZ you can do that by enabling the nvm plugin.

As far as other commands, when you install one zsh needs to refresh the list of available commands. You can do that by running `rehash`.

**Reply by mattdimicelli** [reply]
Thank you so very much!  So, far everything else, does my path variable seem to be configured correctly in ZSH?

**Reply by mcornella** [reply]
Should be yeah, unless you add a new folder where you put commands, like `$HOME/bin` or other common locations, in which case you'd have to add it to `$PATH` (there's a line at the top of the zshrc file you can uncomment). Reply if you need further assistance!

**Reply by mattdimicelli** [reply]
I'm still having some issues.  For example, see this screenshot.
![trash](https://user-images.githubusercontent.com/67718737/111227062-65016900-85b8-11eb-83ea-c7f65aa6b33b.png)

**Reply by mcornella** [reply]
You need to install that command globally with `npm install -g` and probably also add the location where npm installed it to the `$PATH`. Since you're using nvm you might not need to do this last step.

**Reply by mcornella** [reply]
See https://github.com/nvm-sh/nvm/issues/1954

**Reply by miukki** [reply]
nvm install --latest-npm helps
